packages/contracts-bedrock/src/L2/RevenueSharer.sol

```diff
@@ -0,0 +1,175 @@
+     */
+    uint32 public constant WITHDRAWAL_MIN_GAS = 35_000;
+    /**
+     * @dev The percentage coeffieicnt of revenue denominated in basis points that is used in
+     *      Optimism revenue share calculation.
+     */
+    uint256 public constant REVENUE_COEFFICIENT_BASIS_POINTS = 1_500;
+    /**
+     * @dev The percentage coefficient of profit denominated in bass points that is used in
+     *      Optimism revenue share calculation.
+     */
+    uint256 public constant PROFIT_COEFFICIENT_BASIS_POINTS = 250;
+
+    /*//////////////////////////////////////////////////////////////
+                            Immutables
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev The address of the Optimism wallet that will receive Optimism's revenue share.
+     */
+    address payable public BENEFICIARY;
+    /**
+     * @dev The address of the L1 wallet that will receive the OP chain runner's share of fees.
+     */
+    address public L1_WALLET;
+
+    /*//////////////////////////////////////////////////////////////
+                            Events
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Emitted when fees are disbursed.
+     * @param _share The amount of fees shared to the Beneficiary.
+     * @param _total The total funds distributed.
+     */
+    event FeesDisbursed(uint256 _share, uint256 _total);
+    /**
+     * @dev Emitted when fees are received from FeeVaults.
+     * @param _sender The FeeVault that sent the fees.
+     * @param _amount The amount of fees received.
+     */
+    event FeesReceived(address indexed _sender, uint256 _amount);
+
+    /*//////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Constructor for the FeeDisburser contract which validates and sets immutable variables.
+     * @param _beneficiary The address which receives the revenue share.
+     * @param _l1Wallet The L1 address which receives the remainder of the revenue.
+     */
+    constructor(address payable _beneficiary, address payable _l1Wallet) {
+        initialize(_beneficiary, _l1Wallet);
+    }
+
+    function initialize(address payable _beneficiary, address payable _l1Wallet) public initializer {
+        if (_beneficiary == address(0)) revert ZeroAddress("_beneficiary");
+        if (_l1Wallet == address(0)) revert ZeroAddress("_l1Wallet");
+        BENEFICIARY = _beneficiary;
+        L1_WALLET = _l1Wallet;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            External Functions
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Withdraws funds from FeeVaults, sends Optimism their revenue share, and withdraws remaining funds to L1.
+     */
+    function execute() external virtual {
```

Comment:
If you added a mermaid diagram to the specs that showed the flow of calls, that would be amazing